### PR TITLE
Epmldi 1429

### PR DESCRIPTION
**Pull request by tgadiev**
Initial PR.
MySQL 5.5, 5.6, 5,7 and MariaDB supported
CentOS 6 and CentOS 7 supported.

### CODE REVIEW

**Comment by tgadiev** [comment]
Initial PR.
MySQL 5.5, 5.6, 5,7 and MariaDB supported
CentOS 6 and CentOS 7 supported.

File: tasks/clear_innodb_logs.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Stop MySQL to change configs
```
Comment (ViachaslauKabak) [inline tasks/clear_innodb_logs.yml]: All name: values should be in double quotes

File: meta/main.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+dependencies: []
+
+galaxy_info:
+  role_name: "mysql"
+  author: Epam
+  description: MySQL server for RHEL/CentOS.
+  company: "Epam"
```
Comment (ViachaslauKabak) [inline meta/main.yml]: EPAM Systems

File: meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+---
+dependencies: []
+
+galaxy_info:
+  role_name: "mysql"
+  author: Epam
```
Comment (ViachaslauKabak) [inline meta/main.yml]: author: "Lean Delivery team <team@lean-delivery.com>"

File: meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  description: MySQL server for RHEL/CentOS.
+  company: "Epam"
+  license: "Apache"
+  min_ansible_version: 2.4
```
Comment (ViachaslauKabak) [inline meta/main.yml]: 2.5.13?

File: meta/main.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  issue_tracker_url: "https://github.com/lean-delivery/ansible-role-jenkins/issues"
```
Comment (ViachaslauKabak) [inline meta/main.yml]: seems that url is wrong

File: tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
+---
+- name: Check whether the playbook can run on this OS
```
Comment (ViachaslauKabak) [inline tasks/main.yml]: why we need this?

File: vars/RedHat-6-mariadb.yml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+mysql_slow_query_log_file: /var/log/mysql-slow.log
+mysql_log_error: /var/log/mariadb.log
+mysql_syslog_tag: mariadb
+mysql_pid_file: /var/run/mariadb/mariadb.pid
+mysql_config_file: /etc/my.cnf
+mysql_config_include_dir: /etc/my.cnf.d
+mysql_socket: /var/lib/mysql/mysql.sock
+mysql_supports_innodb_large_prefix: False
+
+mysql_version: latest
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mariadb.yml]: should it be in defaults? we cannot define version in playbook

File: vars/RedHat-6-mysqld.yml [hunk 8]
```diff
@@ -0,0 +1,20 @@
+---
+mysql_daemon: mysqld
+mysql_version: 5.7
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mysqld.yml]: should it be in defaults? we cannot define version in playbook

File: vars/RedHat-7-mariadb.yml [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+mysql_slow_query_log_file: /var/log/mysql-slow.log
+mysql_log_error: /var/log/mariadb.log
+mysql_syslog_tag: mariadb
+mysql_pid_file: /var/run/mariadb/mariadb.pid
+mysql_config_file: /etc/my.cnf
+mysql_config_include_dir: /etc/my.cnf.d
+mysql_socket: /var/lib/mysql/mysql.sock
+mysql_supports_innodb_large_prefix: True
+
+mysql_version: latest
```
Comment (ViachaslauKabak) [inline vars/RedHat-7-mariadb.yml]: should it be in defaults? we cannot define version in playbook

File: vars/RedHat-7-mysqld.yml [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+mysql_slow_query_log_file: /var/log/mysql-slow.log
+mysql_log_error: /var/log/mysql.err
+mysql_syslog_tag: mysql
+mysql_pid_file: /var/run/mysqld/mysqld.pid
+mysql_config_file: /etc/my.cnf
+mysql_config_include_dir: /etc/my.cnf.d
+mysql_socket: /var/lib/mysql/mysql.sock
+mysql_supports_innodb_large_prefix: False
+
+mysql_version: 5.7
```
Comment (ViachaslauKabak) [inline vars/RedHat-7-mysqld.yml]: should it be in defaults? we cannot define version in playbook

File: tasks/clear_innodb_logs.yml [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  service:
+    name: '{{ mysql_service_name }}'
+    state: stopped
+
+- name: Wait until the lock file is removed
```
Comment (ViachaslauKabak) [inline tasks/clear_innodb_logs.yml]: Seems that the same was in handlers

**Review by ViachaslauKabak — CHANGES_REQUESTED** [review]

File: tasks/clear_innodb_logs.yml (hunk 11 above)
Comment (tgadiev) [reply tasks/clear_innodb_logs.yml]: Yes, but it's not handler

File: tasks/main.yml (hunk 6 above)
Comment (tgadiev) [reply tasks/main.yml]: It's to correctly handle unsupported OS run

File: tasks/clear_innodb_logs.yml (hunk 11 above)
Comment (mckate) [reply tasks/clear_innodb_logs.yml]: it should be

File: vars/RedHat-6-mariadb.yml (hunk 7 above)
Comment (ViachaslauKabak) [inline vars/RedHat-6-mariadb.yml]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mysqld.yml (hunk 8 above)
Comment (ViachaslauKabak) [inline vars/RedHat-6-mysqld.yml]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mysqld.yml [hunk 12]
```diff
@@ -0,0 +1,20 @@
+---
+mysql_daemon: mysqld
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mysqld.yml]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mysqld.yml [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+mysql_slow_query_log_file: /var/log/mysql-slow.log
+mysql_log_error: /var/log/mysql.err
+mysql_syslog_tag: mysql
+mysql_pid_file: /var/run/mysqld/mysqld.pid
+mysql_config_file: /etc/my.cnf
+mysql_config_include_dir: /etc/my.cnf.d
+mysql_socket: /var/lib/mysql/mysql.sock
+mysql_supports_innodb_large_prefix: False
+
+mysql_repo: http://repo.mysql.com/yum/mysql-{{ mysql_version }}-community/el/6/$basearch/
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mysqld.yml]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mysqld.yml [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+mysql_gpgkey: http://repo.mysql.com/RPM-GPG-KEY-mysql
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mysqld.yml]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mariadb.yml [hunk 15]
```diff
@@ -0,0 +1,22 @@
+---
+mysql_daemon: mariadb
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mariadb.yml]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mariadb.yml [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+mysql_service_name: mysql
+mysql_packages:
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mariadb.yml:7]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mariadb.yml [hunk 17]
```diff
@@ -0,0 +1,22 @@
+---
+mysql_daemon: mariadb
+mysql_service_name: mysql
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mariadb.yml]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mariadb.yml [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+mysql_packages:
+  - MariaDB-client
+  - MariaDB-server
+  - MariaDB-common
+mysql_python_packages:
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mariadb.yml]: If we keep variable here - so it supersedes value in defaults and playbook

File: vars/RedHat-6-mariadb.yml [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+  - MySQL-python
+  - perl-DBD-MySQL
+mysql_slow_query_log_file: /var/log/mysql-slow.log
```
Comment (ViachaslauKabak) [inline vars/RedHat-6-mariadb.yml]: If we keep variable here - so it supersedes value in defaults and playbook

**Review by ViachaslauKabak — CHANGES_REQUESTED** [review]

**Review by ViachaslauKabak — APPROVED** [review]

File: README.md [hunk 20]
```diff
@@ -0,0 +1,185 @@
...
+
+  - Minimal Version of the Ansible for installation: 2.5.13
+  - Supported versions:
+      - Mysql
+          - 5.5
+          - 5.6
+          - 5.7
+      - Mariadb
+          - 10.1
+  - Supported OS:
```
Comment (arno49) [inline README.md:25]: what about RHEL ?

**Review by arno49 — APPROVED** [review]

**Pull request merged**